### ISSUE TITLE
AWS key-pair that is still part of an ASG launch configuration is considered unused

### ISSUE THREAD

**Issue by stepkirk** [cloud-custodian/cloud-custodian#8720]
### Describe the feature

We have a policy that looks for unused AWS EC2 key-pairs. It looks a bit like this:

```
- name: ec2-key-pair-report-unused
    description: |
      Report all unused EC2 Key Pairs
    resource: key-pair
    filters:
      - unused
 ```

I was hoping that the results of this policy would be a list of key-pairs that could be considered for deletion. 
After deleting a couple particularly old ones that hadn't been "used" in awhile, I discovered that they were part of an ASG launch configuration. Since the ASG had a desired capacity of 0 (i.e. no running instances), Custodian didn't consider the key-pair to be in-use.

Could I add something to the above policy to filter out the key-pairs that are configured in ASG launch configs or does the **unused** filter for key-pair need to be updated to consider this?

**Comment by kapilt** [comment]
unused would likely need to be updated, although we could make that an optional configuration item.

**Comment by linuxananth1976** [comment]
@kapilt @stepkirk I'm working on this.  I'll take care it.

**Comment by linuxananth1976** [comment]
worked on [enhancement for asg check too](https://github.com/cloud-custodian/cloud-custodian/pull/8755)

**Comment by linuxananth1976** [comment]
@ajkerrigan ci failed in the PR. #8755

**Comment by ajkerrigan** [comment]
Closed via #8755, thanks @linuxananth1976 ! :tada:
